python/dolfinx/wrappers/caster_mpi.h

```diff
@@ -38,14 +38,23 @@ class type_caster<dolfinx_wrappers::MPICommWrapper>
   // Python to C++
   bool load(handle src, bool)
   {
-    // Simplified version of isinstance(src, mpi4py.MPI.Comm) - avoids
-    // segfault when pybind11 tries to convert some other random type to
-    // MPICommWrapper
-    if (not hasattr(src, "Allgather"))
+    // NOTE: This check would be safer than hasattr but does not evaluate to
+    // true.
+    // if (PyObject_TypeCheck(src.ptr(), &PyMPIComm_Type))
```

Comment:
@chrisrichardson Any idea why this check doesn't work? Perhaps you've been here before.